### PR TITLE
Make `ScopeIdsFixture` compatible with the configuration cache

### PR DESCRIPTION
**Pull request by bamboo**

### CODE REVIEW

**Comment by bamboo** [comment]
@bot-gradle test this

File: subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/ScopeIdsFixture.groovy [hunk 1]
```diff
@@ -92,28 +92,52 @@ class ScopeIdsFixture extends UserInitScriptExecuterFixture {
...
-            def idsFile = file("${normaliseFileSeparators(idsFile.absolutePath)}")
-            if (gradle.parent == null) {
-                idsFile.delete()
+            abstract class CollectScopeIds extends DefaultTask {
+
+                @Internal
+                abstract RegularFileProperty getOutputJsonFile()
+
+                @TaskAction def collect() {
+                    def gradle = services.get(Gradle)
```
Comment (eskatos) [inline subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/ScopeIdsFixture.groovy:104]: Nice trick

File: subprojects/core/src/integTest/groovy/org/gradle/api/tasks/outputorigin/IncrementalBuildOutputOriginIntegrationTest.groovy [hunk 2]
```diff
@@ -38,7 +39,6 @@ class IncrementalBuildOutputOriginIntegrationTest extends AbstractIntegrationSpe
         originBuildInvocationId.originId(taskPath)
     }
 
-    @ToBeFixedForConfigurationCache(because = "build ID doesn't change")
     def "exposes origin build id when reusing outputs"() {
```
Comment (eskatos) [inline subprojects/core/src/integTest/groovy/org/gradle/api/tasks/outputorigin/IncrementalBuildOutputOriginIntegrationTest.groovy:42]: :tada:

**Review by eskatos — APPROVED** [review]
LGTM!

**Comment by bamboo** [comment]
Thanks for the review, @eskatos!

**Pull request merged**